packages/account/src/Sections/Profile/PersonalDetails/personal-details-form.tsx

```diff
@@ -86,29 +89,18 @@ export const PersonalDetailsForm = observer(({ history }: Partial<RouteComponent
         timeout_callback: () => null,
     });
 
-    const isMounted = useIsMounted();
-
     useEffect(() => {
-        if (isMounted()) {
-            const getSettings = async () => {
-                // waits for residence to be populated
-                await WS.wait('get_settings');
-
-                fetchResidenceList?.();
-
-                if (has_residence) {
-                    if (!is_language_changing) {
-                        setIsStateLoading(true);
-                        fetchStatesList().then(() => {
-                            setIsStateLoading(false);
-                        });
-                    }
-                }
-            };
-            getSettings();
+        const init = async () => {
+            await WS.wait('get_settings');
```

Comment:
just a suggestion:
here we are waiting for promises to be resolved one by one, right?
should use here Promise.all() ?

Reply:
async await does the same. The execution control moves to the next line of code only after resolving the function before it

Reply:
yeah, sure @likhith-deriv , my point is just about that, because Promise.all() helps us to run all promises simultaneously, not one by one.
you can ignore my comment.